core/src/main/java/io/questdb/log/LogRollingFileWriter.java

```diff
@@ -369,69 +376,80 @@ private void pushFileStackUp() {
         }
     }
 
-    private void removeExcessiveLogsVisitor(long filePointer, int type) {
-        path.trimTo(logDir.length());
-        path.concat(filePointer).$();
-        logFileName.of(filePointer);
-        if (type == Files.DT_FILE && Chars.contains(logFileName, logFileTemplate) && Files.notDots(filePointer)) {
-            sortedFiles.add(new FileTimestamp(path.toString(), ff.getLastModified(path)));
-        }
-    }
-    
     private void removeExcessiveLogs() {
-        for (int i = 0; i < sortedFiles.size(); i++) {
-            path.of(sortedFiles.get(i).filePath);
-            if ((currentLogSizeSum += Files.length(path)) > nSizeLimit) {
+        path.of(logDir);
+        // The list is sorted by last modification ts ASC, so we iterate in reverse order
+        // starting with the newest files.
+        long totalSize = 0;
+        for (long i = logFileList.size() - 1; i > -1; i -= 2) {
+            final long packedOffsets = logFileList.get(i);
+            final int startOffset = Numbers.decodeLowInt(packedOffsets);
+            final int endOffset = Numbers.decodeHighInt(packedOffsets);
+            CharSequence fileName = logFileNameSink.subSequence(startOffset, endOffset);
+            path.trimTo(logDir.length()).concat(fileName).$();
+            if ((totalSize += Files.length(path)) > nSizeLimit) {
                 if (!ff.remove(path)) {
                     throw new LogError("cannot remove: " + path.$());
```

Comment:
nit: the `.$()` after path, on throw cannot remove, is not necessary.

Reply:
Addressed in 2d4a4dbecca0bc85bf2d84467faf27ff9ab19287